cpp/include/cuml/experimental/fil/detail/raft_proto/buffer.hpp

```diff
@@ -176,21 +184,13 @@ struct buffer {
     swap(first.size_, second.size_);
     swap(first.cached_ptr, second.cached_ptr);
   }
-  buffer<T>& operator=(buffer<T> other)
+  buffer<T>& operator=(buffer<T> const& other)
   {
-    swap(*this, other);
+    auto copy = other;
```

Comment:
Is this not a recursive call?

Reply:
Nope, this is equivalent to:
```
buffer<T> copy{other};
```
That is to say, it invokes the copy constructor, not the copy assignment operator.